File: April 2024 GFG SOLUTION/April-20.py
#User function Template for python3
class Solution:
    #Function to return a list containing the union of the two arrays.
    def findUnion(self, arr1, arr2, n, m):
    
        '''
        :param a: given sorted array a
        :param n: size of sorted array a
        :param b: given sorted array b
        :param m: size of sorted array b
        :return:  The union of both arrays as a list
        '''
        arr1.sort()
        arr2.sort()
        
        ans = []
        
        i = 0
        j = 0
        
        while i < n and j < m:
            if arr1[i] < arr2[j]:
                ans.append(arr1[i])
                i += 1
            elif arr1[i] > arr2[j]:
                ans.append(arr2[j])
                j += 1
            else:
                ans.append(arr1[i])
                i += 1
                j += 1
        
        while i < n:
            ans.append(arr1[i])
            i += 1
        
        while j < m:
            ans.append(arr2[j])
            j += 1
        
        # Remove duplicates
        result = []
        for k in range(len(ans)):
            if k == 0 or ans[k] != ans[k - 1]:
                result.append(ans[k])
        
        return result

        


#{ 
 # Driver Code Starts
#Initial Template for Python 3

# Contributed by : Nagendra Jha
# Modified by : Sagar Gupta


if __name__ == '__main__':
    test_cases = int(input())
    for cases in range(test_cases) :
        n,m = map(int,input().strip().split())
        a = list(map(int,input().strip().split()))
        b = list(map(int,input().strip().split()))
        ob=Solution()
        li = ob.findUnion(a,b,n,m)
        for val in li:
            print(val, end = ' ')
        print()
# } Driver Code Ends

File: April 2024 GFG SOLUTION/April-20.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends


class Solution{
public:
    vector<int> findUnion(int arr1[], int arr2[], int n, int m)
    {
        vector<int> ans;
        sort(arr1, arr1 + n);
        sort(arr2, arr2 + m);
        
        int i = 0, j = 0;
        
        while (i < n && j < m) {
            if (arr1[i] < arr2[j]) {
                ans.push_back(arr1[i]);
                i++;
            } else if (arr1[i] > arr2[j]) {
                ans.push_back(arr2[j]);
                j++;
            } else {
                ans.push_back(arr1[i]);
                i++;
                j++;
            }
        }
        
        while (i < n) {
            ans.push_back(arr1[i]);
            i++;
        }
        
        while (j < m) {
            ans.push_back(arr2[j]);
            j++;
        }
        
        // Remove duplicates
        auto it = unique(ans.begin(), ans.end());
        ans.resize(distance(ans.begin(), it));
        
        return ans; 
    }
};



//{ Driver Code Starts.

int main() {
	
	int T;
	cin >> T;
	
	while(T--){
	    
	    
	    
	    int N, M;
	    cin >>N >> M;
	    
	    int arr1[N];
	    int arr2[M];
	    
	    for(int i = 0;i<N;i++){
	        cin >> arr1[i];
	    }
	    
	    for(int i = 0;i<M;i++){
	        cin >> arr2[i];
	    }
	    Solution ob;
	    vector<int> ans = ob.findUnion(arr1,arr2, N, M);
	    for(int i: ans)cout<<i<<' ';
	    cout << endl;
	    
	}
	
	return 0;
}
// } Driver Code Ends
